src/components/Locations.js

```diff
@@ -0,0 +1,70 @@
+import { Component } from "react";
+import Navbar from "./Navbar";
+
+
+
+class Locations extends Component{
+    constructor(){
+        super()
+        this.state = {
+            display: "Show locations",
+            allLocations: [],
+        };
+    };
+
+    fetchLocations = () => {
+        fetch("https://ghibliapi.herokuapp.com/locations")
+        .then((res) => res.json())
+        .then((data) => {
+            this.setState({
+                allLocations: data,
+
+            });
+        });
+    };
+
+    componentDidMount = () => {
+        this.fetchLocations();
+    };
+
+    handleButtonClick = () => {
+        this.setState({
+            display: "Hide locations"
+        })
+
+    } 
+
+    render(){
+
+        let locationsToDisplay = this.state.allLocations.map((location) => {
+            return(
+                <div className="location-display">
+                    <ul className="location-list">
+                        <li>
+                            <h3>Name: {location.name} </h3>
+                            <h3>Climate: {location.climate} </h3>
+                            <h3>Terrain: {location.terrain} </h3>
+                        </li>
+                    </ul>
+                </div>
+            )
+        })
+
+        return(
+            <div className="locations">
+                <Navbar />
+                <h1>List of Locations</h1>
+                <input
+                    type="button"
+                    value={this.display}
+                    onClick={this.handleButtonClick}
+                />
+                <div className="locations-container" >
+                    { locationsToDisplay }
+                </div>
```

Comment:
```suggestion
                <ul className="locations-container" >
                    { this.state.display === 'Show locations' ? locationsToDisplay : null }
                </ul>
```
Your array of `li`s goes here inside of a `ul` element.

You need to use conditional logic to decide if the locations will show up, or if they'll be hidden. Here, I'm using a **ternary operator**.
This conditional logic isn't perfect, but using the piece of state that you already have (`this.state.display`), this is an idea for how you could go about showing/hiding the locations based on a button click.